wordpress-mobile/WordPress-iOS
# Site Creation : Integrated working Domains suggestion service

### Description
Fixes #10669.

To test:

- Checkout the branch and confirm that tests pass.
- This branch was created off of [this PR branch](https://github.com/wordpress-mobile/WordPress-iOS/pull/10784), so only the last 6 commits (of 12) are relevant to this discussion.
- Step through the site creation flow as shown below and verify that the Domains suggestion service responds to your query.
- Alternately toggle network connectivity and validate the error state is shown as expected.

Update release notes:


### Visual QA

![10669_pr_02](https://user-images.githubusercontent.com/221062/50948366-1c3d4b00-1457-11e9-8bc8-eb91d2df48df.gif)

The error states behave the same way as for Verticals. Static screenshots are included below for reference.

<img width="584" alt="10669_iphone_noconnection" src="https://user-images.githubusercontent.com/221062/50948340-00d24000-1457-11e9-8fe3-fa4fd06655f9.png">

<img width="584" alt="10669_iphone_noresults" src="https://user-images.githubusercontent.com/221062/50948341-00d24000-1457-11e9-8211-37d3063974af.png">
